Report unused import when parent reference is already present

sowmyav24 opened the pull request:
This PR fixes the issue [#405](https://github.com/pinterest/ktlint/issues/405)

- While traversing nodes and building `ref`,  built a set of non redundant imports which omits the imports where parent reference is already present.
- Imports with aliases should contain both name and alias in the importSet
- Static function call and variable have been distinguised

shashachu commented:
@sowmyav24 thanks for the PR! A couple requests:
1) would be great to add a few comments into the code - we're trying to slowly add comments into the standard ruleset since they make great examples for others trying to write their own custom rules
2) I don't know why it wouldn't work, but can you add some tests to ensure auto-formatting works properly for these unused parent imports?

sowmyav24 commented:
@shashachu Thank you for the feedback.
- Added a test to ensure auto formatting works for parent imports.
- Added comments for the functions introduced. Let me know if needs to be more descriptive.

shashachu commented:
Thanks for the fixes!
